assets/src/editor/events/hocs/with-editor-date-entities.js

```diff
@@ -0,0 +1,12 @@
+import { withSelect } from '@wordpress/data';
+import { createHigherOrderComponent } from '@wordpress/compose';
+
+export default createHigherOrderComponent(
+	withSelect( ( select ) => {
+		return {
+			dateEntities: select( 'eventespresso/core' )
+				.getDatetimes(),
+		};
+	} ),
+	'withEditorDateEntities'
```

Comment:
Notice this is simply getting all the datetimes currently in the state and it will always return the complete list when that changes in the current client session.  Think of this replacing the `allDatetimes` prop that was being passed around everywhere.